### DISCUSSION TITLE
How do I make a new window show up on a specific display when using multiple monitors?

### DISCUSSION THREAD

**Discussion by dogou77** [fltk-rs/fltk-rs#1265] [Q&A] [answered]
Hello,

The program I am making with fltk-rs uses multiple windows. 
My goal is to create a window which upon being shown appears on a certain monitor screen.

I have tried using the `.set_screen_num(i32)` before I end and show the window. I have noticed that when I do use that as well as `screen_num()` I am able to get back the screen number I set, but the window will not show up on the specified screen.

Am I using the wrong function to set where the window appears? Is it even possible to tell a window which screen to display on with fltk-rs?

I am using Ubuntu 22.04 LTS with fltk-rs 1.3 and  nightly Rust

Thanks for the help.

**Comment by MoAlyousef** [comment, marked as answer]
Hello

Can you try to check the value of `fltk::app::screen_count()` and report back whether it indeed finds several monitors. You can try then to force the window on a monitor by getting the coordinates of the desired monitor and passing it as coordinates to your window. 
Some of the functions can be found in [examples/screens_info.rs](https://github.com/fltk-rs/fltk-rs/blob/master/fltk/examples/screens_info.rs)

**Reply by dogou77** [reply]
Thanks for replying,

I put `fltk::app::screen_count()` at the beginning of my program and it (correctly) says there are 2 displays.

I will go ahead and try your suggestion of using coordinates to tell the window where to go.

Thanks again.

**Reply by dogou77** [reply]
Using the `.new(i32)` function to get a screen as well as the `.top_left()` function, I was able to find a specific monitor and place the new window at it's coordinates.

Thanks again for your help, I love using FLTK and I am very thankful that you guys brought it over to Rust. Have a nice day.
